src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioSymbolTables.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.PowerFx.Interpreter.Tests.Scenarios
+{
+    // Demonstrate creative use of symbol tables to make a reflection function with a dynamic return type
+    public class ScenarioSymbolTables : PowerFxTest
+    {
+        [Theory]
+        [InlineData("FindRecord(\"Customer\")", typeof(CustomerTable))]
+        [InlineData("FindRecord(\"Vendor\")", typeof(VendorTable))]
+        public void DynamicRecordTypeChecking(string expr, Type expectedType)
+        {
+            var typeCache = new TypeMarshallerCache();
+
+            var config = new PowerFxConfig();
+            config.AddFunction(new FindRecord(typeCache));
+
+            var engine = new RecalcEngine(config);
+
+            var parseResult = engine.Parse(expr);
+            var findRecords = new FindRecordVisitor();
+            parseResult.Root.Accept(findRecords);
+
+            var symbolTable = new SymbolTable();
+            foreach (var tableName in findRecords.RecordTypes)
+            {
+                var record = BaseRecord.CreateRecord(tableName);
+                var recordValue = typeCache.Marshal(record, record.GetType()) as RecordValue;
+
+                symbolTable.AddFunction(new FindRecord(typeCache, recordValue.Type));
```

Comment:
Does this work correctly if there's more than one FindRecord call in a single expression with different args? Our overload resolution is based on types, and I THINK this would always pick the first FindRecord function registered each time

Reply:
That was my assumption that it would work, but I haven't actually tried...

Reply:
I tried, and it doesn't work...
`FindRecord(\"Vendor\").VendorId; FindRecord(\"Customer\").CustomerId;`
results in:
"Name isn't valid. 'CustomerId' isn't recognized."